plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt

```diff
@@ -341,7 +347,12 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
             )
         } catch (e: Exception) {
             LOG.debug { "Creating security scan failed: ${e.message}" }
-            throw codeScanServerException("CreateCodeScanException: " + e.message?.let { it } ?: message("codewhisperer.codescan.run_scan_error_telemetry"))
+            val errorMessage = when {
+                e.message?.contains("Too many requests, please wait before trying again.") == true -> "Too many requests, please wait before trying again."
+                e.message?.contains("Improperly formed request.") == true -> "Improperly formed request."
+                else -> e.message ?: message("codewhisperer.codescan.run_scan_error_telemetry")
```

Comment:
same here for any other expected exceptions from service?

Reply:
I found these from the Kibana. As we get more metrics we need to add if required.